telegram/ext/conversationhandler.py

```diff
@@ -328,5 +334,10 @@ def update_state(self, new_state, key):
             self.conversations[key] = new_state
 
     def _trigger_timeout(self, bot, job):
-        del self.timeout_jobs[job.context]
-        self.update_state(self.END, job.context)
+        self.logger.debug('conversation timeout was triggered!')
+        del self.timeout_jobs[job.context['current_conversation']]
+        handlers = self.states.get(self.TIMEOUT)
+        for candidate in (handlers or []):
+            handler = candidate
+            handler.handle_update(job.context['update'], job.context['dispatcher'])
```

Comment:
We need to run a check_update here. If we don;t it could send an update to a handler not fit for it.